src/main/java/com/jaquadro/minecraft/storagedrawers/client/renderer/TileEntityDrawersRenderer.java

```diff
@@ -513,15 +515,27 @@ private void renderFancyItem(ItemStack itemStack, TileEntityDrawers tile, int sl
         GL11.glDisable(GL11.GL_BLEND);
         GL11.glEnable(GL11.GL_LIGHTING);
 
+        IDrawer drawer = tile.getDrawer(slot);
         try {
-            EntityItem itemEnt = new EntityItem(null, 0, 0, 0, itemStack);
-            itemEnt.hoverStart = 0;
+            EntityItem itemEnt = getCachedEntityItem(drawer, itemStack);
             itemRenderer.doRender(itemEnt, 0, 0, 0, 0, 0);
-        } catch (Exception e) {}
+        } catch (Exception ignored) {}
 
         GL11.glPopMatrix();
     }
 
+    private EntityItem getCachedEntityItem(IDrawer drawer, ItemStack itemStack) {
+        EntityItem cachedEntityItem, newEntityItem;
+        cachedEntityItem = (EntityItem) drawer.getExtendedData(ENT_ITEM_CACHE_KEY);
```

Comment:
if you wanna be efficient during rendering, instead of going through a map why don't you make a new method in the IDrawer interface that returns the cached EntityItem used for rendering

Reply:
returning a pointer to a field is always going to be faster than a map lookup + checking ItemStack.areItemStacksEqual

you can make the interface have a default implementation of returning null for the cached entityItem to render

caching the rendered entity in the extendedData doesn't sound appropriate because that's not extended data that's an internal rendering mechanic

Reply:
I'd say this approach seems fine and seems consistent with how things are already done.  Alex's proposal might be more efficient, but this uses an existing pattern and is already an improvement over what it was before.

Depending on the number of drawers, map access is probably fast enough.  And if profiling shows otherwise, we can improve it then.

Reply:
I profiled it and the map is already 10x faster so thats great, I'm just curious how much better we can do with simple a field